Source/SocketIOClient/Public/SIOMessageConvert.h

```diff
@@ -29,6 +30,10 @@ struct SOCKETIOCLIENT_API FSIOConnectParams
 	/** A map of headers to be added to connection url*/
 	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category = SocketIOConnectionParams)
 	TMap<FString, FString> Headers;
+	
+	/** Optional authorization JSON to send to the server upon initial connection*/
+	UPROPERTY(BlueprintReadWrite, EditAnywhere, Category = SocketIOConnectionParams)
+	USIOJsonValue* Auth;
```

Comment:
likely cleaner if set up as TMap<FString, FString> as base. Follows URL expectation and default params in unreal, objects can't easily be defined in that context, whereas string data can. 
![image](https://user-images.githubusercontent.com/542365/172241703-6766038c-4357-4cfb-b5bd-035f872ebdd5.png)

Reply:
This should be alleviated with the latest commit 66781294996fd0fd736168154adb6564cc1a91bf